scripts/release.sh

```diff
@@ -0,0 +1,114 @@
+then
+	echo "yq tool is required, but could not be found."
+	echo "Install it via: $ brew install yq"
+	exit 1
+fi
+
+# check if `gh` tool is installed.
+if ! command -v gh &> /dev/null
+then
+	echo "Github CLI tool is required, but could not be found."
+	echo "Install it via: $ brew install gh"
+	exit 1
+fi
+
+# check if `gh` tool has auth access.
+# command will return non-zero if not auth'd.
+authd=$(gh auth status -t)
+if [[ $? != 0 ]]
+then
+	echo "ex: $ gh auth login"
+	exit 1
+fi
+
+# check that we're on the `main` branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [ $branch != 'main' ]
+then
+	echo "The 'main' must be the current branch to make a release."
+	echo "You are currently on: $branch"
+	exit 1
+fi
+
+if [ -n "$(git status --porcelain)" ]
+then
+  echo "There are uncommited changes. Please commit and create a pull request or stash them.";
+  exit 1
+fi
+
+version=$(yq eval .version pubspec.yaml)
+echo "Appcues plugin package current version: $version"
+
+# no args, so give usage.
+if [ $# -eq 0 ]
+then
+	echo "Release automation script"
+	echo ""
+	echo "Usage: $ ./release.sh <version>"
+	echo "   ex: $ ./release.sh \"1.0.2\""
+	exit 0
+fi
+
+newVersion="${1}"
+echo "Preparing to release $newVersion..."
+
+versionComparison=$(./scripts/semver.sh $newVersion $version)
+
+if [ $versionComparison != '1' ]
+then
+	echo "New version must be greater than previous version ($version)."
+	exit 1
+fi
+
+read -r -p "Are you sure you want to release $newVersion? [y/N] " response
+case "$response" in
+	[yY][eE][sS]|[yY])
+		;;
+	*)
+		exit 1
+		;;
+esac
+
+# generate the changelog for the git release update
+releaseChangelog=$(git log --pretty=format:"- [%as] %s (%h)" $(git describe --tags --abbrev=0 @^)..@ --abbrev=7 | sed '/[🔧🎬⬆️📸✅💡📝]/d')
+releaseTempFile=$(mktemp)
+echo "$releaseChangelog" >> $releaseTempFile
+
+# generate the updated CHANGELOG.md for the repo (Flutter requirement)
+changelog="## $newVersion\n"
+changelog+=$(git log --pretty=format:"* %s (%h)" $(git describe --tags --abbrev=0 @^)..@ --abbrev=7 | sed '/[🔧🎬⬆️📸✅💡📝]/d')
+changelog+="\n"
+changelogTempFile=$(mktemp)
+echo "$changelog" >> $changelogTempFile
+cat CHANGELOG.md >> $changelogTempFile
+mv $changelogTempFile CHANGELOG.md
+
+# update pubspec.yaml version.
+yq -i e '.version = "'${newVersion}'"' pubspec.yaml
+
+# update example/pubspec.yaml version.
+yq -i e '.version = "'${newVersion}'"' example/pubspec.yaml
+
+# ios/appcues_flutter.podspec - version
+sed -i '' -e "s/$version/$newVersion/g" ios/appcues_flutter.podspec
+
+# android/build.gradle - version
+sed -i '' -e "s/$version/$newVersion/g" android/build.gradle
+
+# commit the version change.
+git commit -am "🔖 Update version to $newVersion"
+git push
+
+# gh release will make both the tag and the release itself.
+gh release create $newVersion -F $releaseTempFile -t $newVersion
+
+# remove the temp files
+rm $releaseTempFile
+rm $changelogTempFile
+
+# dart pub publish
+dart pub publish
```

Comment:
I thought I already left a comment here, but I don't see it, so trying again:

Does it work to use `dart pub token list` to check the pub.dev authentication status? For me I get
```
You do not have any secret tokens for package repositories.
However you can add new tokens using the command below:

    pub token add [hosted-url]
```
so if you get something different, that might work?

Reply:
I get the same thing there, I think the token command may have to do with auth to something other than the main google auth needed for the pub.dev publish? I'm not sure. 

https://dart.dev/tools/pub/cmd/pub-token

> It is used to manage a store of secret tokens for authenticating against third-party servers when [publishing](https://dart.dev/tools/pub/cmd/pub-lish) packages and [retrieving](https://dart.dev/tools/pub/cmd/pub-get) dependencies.

Reply:
Ah makes sense.

You could also add a manual check? like the `Are you sure you want to release $newVersion? [y/N]` just to remind whoever running the release to be sure. But even then, re-running `dart pub publish` if it fails isn't a problem at all since it's the last command in the script.

Reply:
yes that was my thought, can always just re-run the last part as needed to get it published after the release is created.